integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/glue/steps/database/device/DlmsDeviceSteps.java

```diff
@@ -731,4 +719,90 @@ private DeviceModel getDeviceModel(final Map<String, String> inputSettings) {
 
     return null;
   }
+
+  @When("^new keys are registered in the secret management database")
+  public void registerNewKeys(final Map<String, String> inputSettings) {
+    if (!inputSettings.containsKey(PlatformSmartmeteringKeys.DEVICE_IDENTIFICATION)) {
+      throw new IllegalArgumentException("No device identification provided");
+    }
+    final String deviceIdentification =
+        inputSettings.get(PlatformSmartmeteringKeys.DEVICE_IDENTIFICATION);
+    final List<SecretType> secretTypesToCreate =
+        Arrays.asList(E_METER_AUTHENTICATION_KEY, E_METER_ENCRYPTION_KEY_UNICAST);
+    final List<String> keyTypeInputNames =
+        secretTypesToCreate.stream().map(this::getKeyTypeInputName).collect(Collectors.toList());
+    if (Collections.disjoint(inputSettings.keySet(), keyTypeInputNames)) {
+      throw new IllegalArgumentException(
+          "None of the following keys provided: " + keyTypeInputNames.toString());
+    }
+    final DbEncryptionKeyReference encryptionKeyRef =
+        this.encryptionKeyRepository
+            .findByTypeAndValid(EncryptionProviderType.JRE, new Date())
+            .iterator()
+            .next();
+    for (int i = 0; i < secretTypesToCreate.size(); i++) {
+      if (inputSettings.containsKey(keyTypeInputNames.get(i))) {
+        final DbEncryptedSecret secret =
+            new SecretBuilder()
+                .withDlmsDevice(new DlmsDevice(deviceIdentification))
+                .withSecretType(secretTypesToCreate.get(i))
+                .withKey(inputSettings.get(keyTypeInputNames.get(i)))
+                .withSecretStatus(SecretStatus.NEW)
+                .build();
+        secret.setEncryptionKeyReference(encryptionKeyRef);
+        this.encryptedSecretRepository.save(secret);
+      }
+    }
+  }
+
+  @Then("after 15 seconds, the new keys are recovered")
+  public void newKeysAreRecovered(final Map<String, String> inputSettings)
+      throws InterruptedException {
+    final long waitTimeInMillis = 15 * 1000;
+    if (!inputSettings.containsKey(PlatformSmartmeteringKeys.DEVICE_IDENTIFICATION)) {
+      throw new IllegalArgumentException("No device identification provided");
+    }
+    final String deviceIdentification =
+        inputSettings.get(PlatformSmartmeteringKeys.DEVICE_IDENTIFICATION);
+    if (!inputSettings.containsKey(KEY_DEVICE_AUTHENTICATIONKEY)
+        && !inputSettings.containsKey(KEY_DEVICE_ENCRYPTIONKEY)) {
+      throw new IllegalArgumentException("No authentication or encryption key provided");
+    }
+    Thread.sleep(waitTimeInMillis);
```

Comment:
Can't be be done with a _Wait.until_ function built in in the cucumber test enviroment. This might save some time if the keys are recovered in less than 15 seconds

Reply:
Yes, replaced it with Wait.until